pkg/function-proxy/proxy.go

```diff
@@ -80,10 +83,35 @@ func startNativeDaemon() {
 	}
 }
 
+func gracefulShutdown(server *http.Server) {
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
as far as I know, you cannot capture a `SIGTERM` signal

Reply:
Actually you can capture SIGTERM, and it's what kubernetes sends to a pod when it requests a graceful shutdown. 30 seconds later it'll send SIGKILL which can't be captured, maybe that's what you're thinking of?

It can be tested by running the proxy locally and running `kill -15 <pid>`.

Reply:
See https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-terminating-with-grace

Reply:
I see, indeed they suggest so. I tried it with a simple example and it didn't seem to work:

```
▶ cat t.go 
package main

import (
    "fmt"
    "os"
    "os/signal"
    "syscall"
)

func main() {

    sigs := make(chan os.Signal, 1)
    done := make(chan bool, 1)

    signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)

    go func() {
        sig := <-sigs
        fmt.Println()
        fmt.Println(sig)
        done <- true
    }()

    fmt.Println("awaiting signal")
    <-done
    fmt.Println("exiting")
}

▶ go run t.go
awaiting signal
^C
interrupt
exiting

▶ go run t.go  
awaiting signal
[1]    17141 terminated  go run t.go
```

As you can see, in the second execution I am sending `kill -15 <pid>` and it's not catching the signal (but I may be missing something),

Reply:
Curiously I tried your example.
When running with `go run foo.go` it seems to make two processes, when I sent sigterm to the binary in /tmp it worked as expected.
i.e. the second entry in this `ps` output
![image](https://user-images.githubusercontent.com/8149130/72522922-d5c4cc80-38b2-11ea-9b52-9aade0c2aee9.png)
Compiling with `go build foo.go` and doing the same with the produced binary should work as well.

Anyway thanks for being so responsive!
I hope to get the node runtime doing the same soon.

Reply:
Oh, I see, good to know, thanks for following up!